scene/gui/dialogs.cpp

```diff
@@ -286,19 +286,26 @@ void AcceptDialog::_custom_action(const String &p_action) {
 	custom_action(p_action);
 }
 
+void AcceptDialog::_custom_button_visibility_changed(Button *button, Control *spacer) {
+	spacer->set_visible(button->is_visible());
+}
+
 Button *AcceptDialog::add_button(const String &p_text, bool p_right, const String &p_action) {
 	Button *button = memnew(Button);
 	button->set_text(p_text);
 
+	Control *spacer;
```

Comment:
Could be good to store a hashmap of buttons and spacers. Will make `remove_button`'s logic safer as well.

Reply:
The spacer can also be stored in button's metadata. It'd be simpler 🙃

Reply:
Went for metadata, additional HashMap seems excessive to me. Maybe it would make sense if we'd want to add a method like `get_buttons` or something like that. :thinking: But that's out of scope of this PR.